### PR TITLE
PPTP-1204 - allow enrolled user to access confirmation page

### PR DESCRIPTION
**Pull request by SteveSugden**
- solves the potential race condition where subscription request completes before confirmation page is displayed
- creates a second version of AuthAction which does not check enrolments

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/SignOutController.scala [hunk 1]
```diff
@@ -16,19 +16,19 @@
...
 import uk.gov.hmrc.plasticpackagingtax.registration.views.html.session_timed_out
 import uk.gov.hmrc.plasticpackagingtax.registration.views.model.SignOutReason
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 
-import javax.inject.Inject
 import scala.concurrent.Future
 
 class SignOutController @Inject() (
-  authenticate: AuthAction,
+  authenticate: AuthAllowEnrolmentAction,
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/SignOutController.scala:31]: Good spot.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/actions/AuthAction.scala [hunk 2]
```diff
@@ -41,7 +41,31 @@ class AuthActionImpl @Inject() (
   metrics: Metrics,
   mcc: MessagesControllerComponents,
   appConfig: AppConfig
-) extends AuthAction with AuthorisedFunctions {
+) extends AuthActionBase(authConnector, allowedUsers, metrics, mcc, appConfig) with AuthAction {
+  override val checkAlreadyEnrolled: Boolean = true
+}
+
+class AuthAllowEnrolmentActionImpl @Inject() (
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/actions/AuthAction.scala:48]: `AuthActionNoEnrolmentCheckAction`?

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
Suggesting a minor name change.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/actions/AuthAction.scala (hunk 2 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/actions/AuthAction.scala]: Agreed and changed

**Review by RyanKellyHMRC — APPROVED** [review]

**Pull request merged**